spec/amp-cache-debugging.md

```diff
@@ -0,0 +1,24 @@
+# AMP Cache Debugging
+
+**Why is my doc broken on the AMP cache?**
```

Comment:
maybe phrase "on the AMP cache" as "when served from an AMP cache" instead.

Would be good to phrase throughout as general AMP caches, rather than "the" AMP cache, so the guidelines aren't Google-specific